straxbra/plugins.py

```diff
@@ -356,33 +341,14 @@ def compute(self, peaks):
             results[p_i] = tuple(self.reconstructed_position(apc))
         return results
 
-    # Function to return non-negative value corresponding to (radial position - radius TPC) if inside TPC,
-    # used for constraints
-    @staticmethod
-    def insidevolume(inputs):
-        p_TPC_radius = 35
-        return (p_TPC_radius - np.hypot(inputs[0], inputs[1]))
-    
-    # Radial LRF model
-    # from: Position Reconstruction in a Dual Phase Xenon Scintillation Detector (https://arxiv.org/abs/1112.1481)
-    @staticmethod
-    def eta(r, A, r0, a, b, alpha):
-        return A * np.exp( - a * (r/r0) / (1 + (r/r0) ** (1 - alpha)) - b / (1 + (r/r0) ** (- alpha)))   
-
-    ## Position dependent LRF values individual PMTs from model
-    def LRF_PMTs(self, x, y):
-        LRF_PMTs_array = np.array([(self.eta(np.sqrt((x - self.pmt_positions[j, 0])**2 + (y - self.pmt_positions[j, 1])**2), *self.fitparameters[j])) for j in range(0,(self.pmt_positions).shape[0])])
-        return LRF_PMTs_array / (np.sum(LRF_PMTs_array))
-    
     ## Reconstruct position inside S2 region
     def reconstructed_position(self, input_array):
+        ## Normalize sum input to 1 in order to correspond to area fraction in top array
         HFs_input = input_array / np.sum(input_array)
-        reconstruct = lambda x: np.sum(((self.LRF_PMTs(x[0], x[1]) - HFs_input)**2) / (self.LRF_PMTs(x[0], x[1])))
-        x0 = [0.001,0.001]
-        meth = 'SLSQP'
-        cons = ({'type': 'ineq', "fun": self.insidevolume})
-        res = minimize(reconstruct, x0, method=meth, constraints=cons)
-        return res.x
+        ## Use model to reconstruct position
+        ## Important: Factor 70 for rescaling label
+        predictions = self.model_NN.predict(np.array([HFs_input]))[0]*70
```

Comment:
The labels the NN was trained on (i.e. the x and y positions) were scaled to [-0.5, 0.5] in order to improve the efficiency and accuracy, so using this trained model we have to rescale to the physical positions.

Reply:
For future work it may be better to scale the training data to [-1,1] because then your scaling factor is the TPC radius, which we would specify as an option.

Reply:
In that sense the current scaling factor is actually the TPC diameter, so if you want it's possible to specify it as an option.